src/blocks/scratch3_motion.js

```diff
@@ -52,6 +55,41 @@ Scratch3MotionBlocks.prototype.pointInDirection = function (args, util) {
     util.target.setDirection(args.DIRECTION);
 };
 
+Scratch3MotionBlocks.prototype.glide = function (args, util) {
+    if (!util.stackFrame.timer) {
+        // First time: save data for future use.
+        util.stackFrame.timer = new Timer();
+        util.stackFrame.timer.start();
+        util.stackFrame.duration = Cast.toNumber(args.SECS);
+        util.stackFrame.startX = util.target.x;
+        util.stackFrame.startY = util.target.y;
+        util.stackFrame.endX = Cast.toNumber(args.X);
+        util.stackFrame.endY = Cast.toNumber(args.Y);
+        if (util.stackFrame.duration <= 0) {
+            // Duration too short to glide.
+            util.target.setXY(util.stackFrame.endX, util.stackFrame.endY);
+            return;
+        }
+        util.yieldFrame();
```

Comment:
Actually, after looking at it further, I think this is right:
https://scratch.mit.edu/projects/120528894/#editor

`yieldFrame` here means pause the thread's execution for a frame and come back to the block on the next frame (*not* the same as `interp.redraw` in 2.0).

I think if we moved it up to even yield when glide 0s, the behavior would be differ from 2.0. In the above project, 3 frames would pass before the turn.